pyidr/study_parser.py

```diff
@@ -205,138 +222,116 @@ def get_lines(self, index, component_regexp):
-            })
-        return
-
-    def parse_publications(self):
-
-        titles = self.study['Study Publication Title'].split('\t')
-        authors = self.study['Study Author List'].split('\t')
+        titles = self.study.get('Study Publication Title', [])
+        authors = self.study.get('Study Author List', [])
         assert len(titles) == len(authors), (
             "Mismatching publication titles and authors")
-        if titles == [''] and authors == ['']:
-            return
-
-        publications = [{"Title": title, "Author List": author}
-                        for title, author in zip(titles, authors)]
-
-        def parse_ids(key, pattern):
-            if key not in self.study:
-                return
-            split_ids = self.study[key].split('\t')
-            key2 = key.strip("Study ")
-            for i in range(len(split_ids)):
-                if not split_ids[i]:
-                    continue
-                m = pattern.match(split_ids[i])
-                if not m:
-                    raise Exception("Invalid %s: %s" % (key2, split_ids[i]))
-                publications[i][key2] = m.group("id")
-
-        parse_ids("Study PubMed ID", re.compile(r"(?P<id>\d+)"))
-        parse_ids("Study PMC ID", re.compile(r"(?P<id>PMC\d+)"))
-        parse_ids("Study DOI", DOI_PATTERN)
-
-        self.study["Publications"] = publications
 
-    @staticmethod
-    def parse_data_doi(d, key):
-        if key not in d:
-            return {}
-        m = DOI_PATTERN.match(d[key])
-        if not m:
-            raise Exception(
-                "Invalid Data DOI: %s" % d[key])
-        return {"Data DOI": m.group("id")}
+        for doi in self.study.get("Study DOI", []):
+            validate_doi(doi)
 
-    @staticmethod
-    def parse_organism(component):
-        key = '%s Organism' % component['Type']
-        if 'Study Organism' in component and key in component:
-            raise Exception("Organism defined both at the study and %s level"
-                            % component['Type'])
-        elif 'Study Organism' not in component and key not in component:
-            raise Exception("Missing organism field")
-        elif 'Study Organism' in component:
-            component[key] = component['Study Organism']
-        return
+        for pubmed_id in self.study.get("Study PubMed ID", []):
+            pattern = re.compile(r"(?P<id>\d+)")
+            if pubmed_id != '' and not pattern.match(pubmed_id):
+                raise Exception("Invalid PubMed ID: %s" % pubmed_id)
 
-    def get_study_accession(self):
-        return self.study[r"Comment\[IDR Study Accession\]"]
+        for pmc_id in self.study.get("Study PMC ID", []):
+            pattern = re.compile(r"(?P<id>PMC\d+)")
+            if pmc_id != '' and not pattern.match(pmc_id):
+                raise Exception("Invalid PMC ID: %s" % pmc_id)
 
-    def get_study_name(self):
+    def validate_organisms(self):
```

Comment:
Similarly, if this util were to just generate a JSON representation that could have a validation applied, this might night need to be imperatively coded.

Reply:
Also agreed at least for the simple validation. I am unsure of how much of the validation across keys can be stored in a simple schema but I feel I need to read more
In many ways, this brings back questions we have been raising as part of the OME-NGFF validation work. It would be interesting to keep these two efforts in line and use similar strategies/decisions in different contexts.